test/e2e/cloud-provider-azure.go

```diff
@@ -40,75 +36,42 @@ const (
 	azureDiskCSIDriverHelmReleaseName = "azuredisk-csi-driver-oot"
 )
 
-func InstallHelmCharts(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, result *clusterctl.ApplyClusterTemplateAndWaitResult){
-	InstallCloudProviderAzureHelmChart(ctx,input,result)
-	InstallAzureDiskCSIDriverHelmChart(ctx,input,result)
-}
-
 // InstallCloudProviderAzureHelmChart installs the official cloud-provider-azure helm chart
-// Fulfills the clusterctl.Waiter type so that it can be used as ApplyClusterTemplateAndWaitInput data
-// in the flow of a clusterctl.ApplyClusterTemplateAndWait E2E test scenario
-func InstallCloudProviderAzureHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, result *clusterctl.ApplyClusterTemplateAndWaitResult) {
-	By(fmt.Sprintf("Ensuring the kubeconfig secret for cluster %s/%s exists before installing cloud-provider-azure components", input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName))
-	WaitForWorkloadClusterKubeconfigSecret(ctx, input)
+// and validates that expected pods exist and are Ready.
+func InstallCloudProviderAzureHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput) {
 	By("Installing the correct version of cloud-provider-azure components via helm")
 	values := []string{fmt.Sprintf("infra.clusterName=%s", input.ConfigCluster.ClusterName)}
 	InstallHelmChart(ctx, input, cloudProviderAzureHelmRepoURL, cloudProviderAzureChartName, cloudProviderAzureHelmReleaseName, values)
 	By("Waiting for a Running cloud-controller-manager pod")
 	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
 	workloadClusterClient := clusterProxy.GetClient()
-	cloudControllerManagerPodLabel, err := labels.Parse("component=cloud-controller-manager")
```

Comment:
This manual pod filter and list is too cute by half, and in addition it is less accurate in terms of pod count than relying upon the parent Deployment resource metadata (see the new `WaitForDeployment` helper func).